extensions/scarb-doc/src/lib.rs

```diff
@@ -29,16 +29,29 @@ pub struct AdditionalMetadata {
 pub fn generate_packages_information(
     metadata: &Metadata,
     metadata_for_packages: &[PackageMetadata],
+    document_private_items: bool,
 ) -> Vec<PackageInformation> {
     let mut packages_information = vec![];
     for package_metadata in metadata_for_packages {
         let authors = package_metadata.manifest_metadata.authors.clone();
+        let edition = package_metadata
+            .edition
+            .as_ref()
+            .map(|edition| edition_from_string(edition).unwrap());
+
+        let should_ignore_visibility = match edition {
+            Some(edition) => edition.ignore_visibility(),
+            None => Edition::latest().ignore_visibility(),
```

Comment:
```suggestion
            None => Edition::default().ignore_visibility(),
```
If not specified, we use the `default` edition, which is not the same as `latest` (for compatibility with packages older than compiler editions). (Generally it should also only be `None` for Scarb older than introduction of editions.)